modules/standard/DynamicIters.chpl

```diff
@@ -154,6 +154,113 @@ where tag == iterKind.follower
   }
 }
 
+//************************* Multidimensional Dynamic iterator
+/*
+
+  :arg c: The domain to iterate over. The rank of the domain must be greater 
+          than one. If using a 1d domain, the user should use the existing
+          dynamic iterators (use DynamicIters). 
+  :type c: `domain`
+
+  :arg chunkSize: The size of chunks to be yielded to each thread. Must be
+                  greater than zero
+  :type chunkSize: `int`
+
+  :arg numTasks: The number of tasks to use. Must be >= zero. If this argument
+                 has the value 0, it will use the value indicated by
+                 ``dataParTasksPerLocale``.
+  :type numTasks: `int`
+
+  :arg parDim: The index of the dimension to parallelize across. Must be > 0.
+                Must be <= the rank of the domain ``c``. Defaults to 1. 
+  :type parDim: `int`
+
+  :yields: Slices of the domain ``c``
+
+  Given an input domain ``c``, each task is assigned slices of ``c``. The
+  chunks each have ``chunkSize`` slices in them (or the remaining iterations
+   if there are fewer than ``chunkSize``). This continues until there are no 
+   remaining iterations in the dimension of ``c`` indicated by ``parDim``. 
+
+  LIMITATIONS:
+  user can select parDim, but only if they also define numTasks (even to 0)
+    should parDim be a non-default param?
+  explicitly types the ranges (so they will likely only work with ints)
+  May or may not properly lead/follow with other domain iterators
```

Comment:
Two things:

You can name the argument 'parDim' to give it a value, like this:

```chpl
var dom = {1..5, 1..10};
forall i in dynamic(dom, 1, parDIm=2) do ...
```

Second, I think by the end of this you should be able to say that this iterator should lead and follow with rectangular domains correctly, and with any domains that interact nicely with rectangular domains.